### ISSUE TITLE
Cannot load pretrained CIFAR-10 checkpoints

### ISSUE THREAD

**Issue by ankitkv** [tml-epfl/sharpness-vs-generalization#1]
Hi!

When trying to load the pretrained checkpoints provided in the Google Drive, I get an error when running `eval_sharpness_final.sh` for the ViTs (I have not tried ResNets). For instance, for the file
`2023-01-14 23:40:10.063 dataset=cifar10 model=vit_exp epochs=200 lr_max=0.155923 model_width=512 l2_reg=0.0 sam_rho=0.0 batch_size=128 frac_train=1 p_label_noise=0.0 lr_schedule=cyclic augm=False randaug=False seed=0 epoch=200.pth`, I get the following error:
```
Traceback (most recent call last):
  File "sharpness-vs-generalization/eval_sharpness.py", line 60, in <module>
    model.load_state_dict({k: v for k, v in model_dict.items()})
  File ".conda/envs/sharpness/lib/python3.10/site-packages/torch/nn/modules/module.py", line 2041, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for SimpleViT:
	Missing key(s) in state_dict: "to_patch_embedding.2.weight", "to_patch_embedding.2.bias", "to_patch_embedding.3.weight", "to_patch_embedding.3.bias", "transformer.norm.weight", "transformer.norm.bias", "linear_head.weight", "linear_head.bias".
	Unexpected key(s) in state_dict: "linear_head.0.weight", "linear_head.0.bias", "linear_head.1.weight", "linear_head.1.bias".
	size mismatch for to_patch_embedding.1.weight: copying a param with shape torch.Size([512, 48]) from checkpoint, the shape in current model is torch.Size([48]).
	size mismatch for to_patch_embedding.1.bias: copying a param with shape torch.Size([512]) from checkpoint, the shape in current model is torch.Size([48]).
```

Is there a particular version of vit-pytorch I need? Thank you!

**Comment by max-andr** [comment]
Hey,

So I used the most recent version of `vit-pytorch` when I did these experiments at the beginning of January 2023. It should be this one: https://github.com/lucidrains/vit-pytorch/tree/500e23105a294b55a585462deab1884af264888a. Could you try to load the weights with it?

It seems that there were subsequent commits in the library that might have affected SimpleViT.

Best,
Maksym

**Comment by ankitkv** [comment]
Thanks! I was able to fix this by using `vit-pytorch<1.0.0` in the `requirements.txt`. 1.0 onwards breaks your code. I'll leave this issue open so you can choose to close it now or after you update the version restriction.

**Comment by max-andr** [comment]
Thanks for the comment! I'll change the requirements.txt.
